src/app/page/Missions/Missions.jsx

```diff
@@ -26,9 +28,31 @@ const PlaceHolder = styled.div`
 const MissionsPage = ({ user, history, firebase, ...rest }) => {
   const missions = useSelector((state) => state.firestore.ordered.missionsTodo);
   const firestore = useFirestore();
+  const [popupOpen, setPopupOpen] = useState(false);
+  const [phoneNumber, updatePhoneNumber] = useState(null);
 
+  function refreshPage() {
+    window.location.reload();
+  }
+  function handleLinkPhoneNumberChange(e) {
+    updatePhoneNumber(e.target.value);
+  }
+
+  function handleLinkPhoneLinkButtonClick() {
+    if (phoneNumber) {
+      var appVerifier = new firebase.auth.RecaptchaVerifier("recaptcha", { size: "small" });
+      User.linkPhoneAuthentication(firebase, phoneNumber, appVerifier, refreshPage);
+    }
+  }
+  function handlePopupClose() {
+    setPopupOpen(false);
+  }
   function volunteerForMission(missionId) {
-    User.assignAsVolunteer(firestore, missionId, user.uid);
+    if (!user.phoneNumber) {
+      setPopupOpen(true);
```

Comment:
Actually, I have done some linking work already in the user profile page. Maybe instead of doing a popup just redirect them to user/profile if they are login but not have a number and say: Hey, add your phone number here

If they are not login, then -> login page